state-sync/inter-component/consensus-notifications/src/lib.rs

```diff
@@ -341,8 +341,8 @@ mod tests {
         );
 
         // Verify the notification arrives at the receiver
-        match consensus_listener.notification_receiver.try_next() {
-            Ok(Some(consensus_notification)) => match consensus_notification {
+        match consensus_listener.select_next_some().now_or_never() {
```

Comment:
lol now or never.  Looks like you moved the stream off notification receiver to the listener.  nice